pallets/grants/src/lib.rs

```diff
@@ -214,6 +227,91 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Initiate a bridge transfer of all vested funds to the given `eth_address` on the given `chain_id`.
+		/// This process will need to be completed by the bridge oracles.
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::bridge_all_vesting_schedules())]
+		pub fn bridge_all_vesting_schedules(
+			origin: OriginFor<T>,
+			eth_address: [u8; 20],
+			bridge_id: u32,
+		) -> DispatchResultWithPostInfo {
+			let from = ensure_signed(origin)?;
+			ensure!(
+				Bridges::<T>::contains_key(&BridgeId(bridge_id)),
+				Error::<T>::BridgeNotFound
+			);
+
+			ensure!(!Self::renounced(from.clone()), Error::<T>::Renounced);
+			let locked_amount_left = Self::do_claim(&from);
+			if locked_amount_left.is_zero() {
+				<VestingSchedules<T>>::remove(from);
+				Self::deposit_event(Event::NoVestedFundsToBridgeAfterClaim);
+				return Ok(().into());
+			}
+
+			let free_balance = T::Currency::free_balance(&from);
+			let bridgeable_funds = locked_amount_left.min(free_balance);
+
+			T::Currency::remove_lock(VESTING_LOCK_ID, &from);
+			T::Currency::settle(
+				&from,
+				T::Currency::burn(bridgeable_funds),
+				WithdrawReasons::RESERVE,
+				ExistenceRequirement::AllowDeath,
+			)
+			.map_err(|_| Error::<T>::FailedToSettleBridge)?;
+
+			let grants = <VestingSchedules<T>>::take(&from);
+
+			Self::deposit_event(Event::BridgeInitiated {
+				to: eth_address,
+				bridge_id,
+				amount: bridgeable_funds,
+				grants,
+			});
+
+			Ok(().into())
+		}
+
+		/// Allow governance to indicate a bridge is setup between the parachain and a remote chain.
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::set_bridge())]
+		pub fn set_bridge(
+			origin: OriginFor<T>,
+			bridge_id: u32,
+			bridge_name: Vec<u8>,
```

Comment:
Why is `bridge_name` necessary? Just UX?
We could also simplify towards chain ID being the universal ID to manage bridges

Reply:
1. bridge_name will improve the UX a lot but only occupies 32 bytes.
2. Can you explain your second point more?